src/Common/ThreadProfileEvents.cpp

```diff
@@ -518,13 +514,17 @@ void PerfEventsCounters::finalizeProfileEvents(PerfEventsCounters & counters, Pr
             continue;
 
         const PerfEventInfo & info = raw_events_info[i];
-        const PerfEventValue & raw_value = counters.raw_event_values[i];
-
-        profile_events.increment(info.profile_event, raw_value.value);
-        if (info.profile_event_running.has_value())
-            profile_events.increment(info.profile_event_running.value(), raw_value.time_running);
-        if (info.profile_event_enabled.has_value())
-            profile_events.increment(info.profile_event_enabled.value(), raw_value.time_enabled);
+        const PerfEventValue & old_value = old_values[i];
+        const PerfEventValue & new_value = counters.raw_event_values[i];
+
+        // Account for counter multiplexing. time_running/time_enabled are
+        // not reset by PERF_EVENT_IOC_RESET, so we have to calculate deltas
+        // from old values.
+        profile_events.increment(info.profile_event,
+            (new_value.value - old_value.value)
+                * (new_value.time_running - old_value.time_running)
```

Comment:
`value` is a number of raw events and `time_running` is always <= `time_enabled`; so, to get the scaled final result you should use a formulae `final_count = raw_count * time_enabled/time_running` instead of the current 
`final_count = raw_count * time_running/time_enabled`:
docs: https://perf.wiki.kernel.org/index.php/Tutorial > "multiplexing and scaling events"